### PR TITLE
10232 oal editor enh

### PR DESCRIPTION
**Pull request by keithbrown**

### CODE REVIEW

File: doc-bridgepoint/notes/10232_oal_editor_enh/10232_oal_editor_enh_dnt.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+max values (e.g. `[0,11]`).  European users read this as a North American would read a decimal number that
+uses `.`.  We shall change the range separator to use `..` instead (e.g. `[0..11]`) [2.2].   
+ 
+### 4. Requirements
+
+4.1 OAL editor shall properly honor the tab-width preference.   
+4.1.1  When tabs are specified to be used, tabs shall follow the width specified in the preference  
+4.1.2  When spaces are specified to be used, the correct number of spaces shall be inserted to match the tab width  
+     
+4.2 Ranges shall be displayed on the canvas with `..` instead of `,`
```
Comment (rmulvey) [inline doc-bridgepoint/notes/10232_oal_editor_enh/10232_oal_editor_enh_dnt.md]: This really does not belong in this document.

File: src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        } catch (BadLocationException ble) {
+            CorePlugin.logError("Invalid edit location exception caught.", ble);
+        }
+    }
+
+    public static int findEndOfWhiteSpace(IDocument document, int offset, int end) throws BadLocationException {
+        while (offset < end) {
+            char c = document.getChar(offset);
+            if (c != ' ' & c != '\t') {
```
Comment (rmulvey) [inline src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java]: This leaves out CR/LF. Perhaps there is a better way to do this. There is a function available to check for whitespace.

File: src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                return offset;
+            }
+            offset++;
+        }
+        return end;
+    }
+
+    public static String getIndentOfLine(IDocument document, int line) throws BadLocationException {
+        if (line > -1) {
+            int start = document.getLineOffset(line);
```
Comment (rmulvey) [inline src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java]: I am not confident this is properly handlng newline.  ie CR LF. Looks like a place where OS-specific could be present.

File: src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            return "";
+        }
+    }
+
+    public static String getAdditionalIndent(IDocument document, int line) throws BadLocationException {
+        if (line > -1) {
+            int start = document.getLineOffset(line);
+            int end = start + document.getLineLength(line) - 1;
+            String text = document.get(start,end-start);
+            if ( text.matches("^\\s*(?i)(if|elif|while)\\s*\\(?.*") || text.matches("^\\s*(?i)(else|for\\s+each\\s+).*") ) {
```
Comment (rmulvey) [inline src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java]: Add a comment here to explain the reg expression

Comment (keithbrown) [reply src/org.xtuml.bp.ui.text/src/org/xtuml/bp/ui/text/editor/oal/OALAutoEditStrategy.java]: added

File: doc-bridgepoint/notes/10232_oal_editor_enh/10232_oal_editor_enh_dnt.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+The OAL editor is also not good at honoring the user preferences for tab width and whether spaces or
+tabs should be used for indentation [2.4].   
+
+Currently UDTs with a specified range are showed on the canvas using the comma separator between min and 
+max values (e.g. `[0,11]`).  European users read this as a North American would read a decimal number that
+uses `.`.  We shall change the range separator to use `..` instead (e.g. `[0..11]`) [2.2].   
+ 
+### 4. Requirements
+
```
[the remark this answers is not in the document]
Comment (keithbrown) [reply doc-bridgepoint/notes/10232_oal_editor_enh/10232_oal_editor_enh_dnt.md]: fixed

**Pull request merged**